// @builder.io/react/dist/lib/src/blocks/raw/RawText.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = __importStar(require("react"));
var sdk_1 = require("@builder.io/sdk");
exports.RawText = function (props) {
    var attributes = props.attributes || {};
    return (React.createElement("span", { className: (attributes === null || attributes === void 0 ? void 0 : attributes.class) || (attributes === null || attributes === void 0 ? void 0 : attributes.className), dangerouslySetInnerHTML: { __html: props.text || '' } }));
};
sdk_1.Builder.registerComponent(exports.RawText, {
    name: 'Builder:RawText',
    hideFromInsertMenu: true,
    inputs: [
        {
            name: 'text',
            bubble: true,
            type: 'longText',
            required: true,
        },
    ],
});
